src/lib.rs

```diff
@@ -2729,6 +2729,145 @@ pub fn is_maximal_matching(
     })
 }
 
+/// Return the core number for each node
+///
+/// A k-core is a maximal subgraph that contains nodes of degree k or more.
+///
+/// :param PyGraph: The graph to get core numbers
+///
+/// :returns: A dictionary keyed by node index to the core number
+/// :rtype: dict
+#[pyfunction]
+#[text_signature = "(graph, /)"]
+pub fn graph_core_number(
+    py: Python,
+    graph: &graph::PyGraph,
+) -> PyResult<PyObject> {
+    let mut cores: HashMap<NodeIndex, usize> =
+        HashMap::with_capacity(graph.node_count());
+    let mut node_vec: Vec<NodeIndex> = graph.graph.node_indices().collect();
+    let mut degree_map: HashMap<NodeIndex, usize> =
+        HashMap::with_capacity(graph.node_count());
+    let mut nbrs: HashMap<NodeIndex, Vec<NodeIndex>> =
+        HashMap::with_capacity(graph.node_count());
+    let mut node_pos: HashMap<NodeIndex, usize> =
+        HashMap::with_capacity(graph.node_count());
+
+    for k in node_vec.iter() {
+        let ki = (*k).index();
+        cores.insert(*k, graph.degree(ki));
+        degree_map.insert(*k, graph.degree(ki));
+        nbrs.insert(*k, graph.graph.neighbors(*k).collect());
+    }
+    node_vec.par_sort_by_key(|k| degree_map.get(k));
+
+    let mut bin_boundaries = vec![0];
+    let mut curr_degree = 0;
+    for (i, v) in node_vec.iter().enumerate() {
+        node_pos.insert(*v, i);
+        let v_degree = degree_map[v];
+        if v_degree > curr_degree {
+            for _ in 0..v_degree - curr_degree {
+                bin_boundaries.push(i);
+            }
+            curr_degree = v_degree;
+        }
+    }
+
+    for v_ind in 0..node_vec.len() {
+        let v = node_vec[v_ind];
+        let v_nbrs = nbrs[&v].clone();
+        for u in v_nbrs {
+            if cores[&u] > cores[&v] {
+                nbrs.get_mut(&u).unwrap().retain(|&n| n == v);
```

Comment:
```suggestion
                nbrs.get_mut(&u).unwrap().retain(|&n| n != v);
```

Reply:
Thanks! Not sure how the tests worked with this mistake... I guess more tests are needed